### DISCUSSION TITLE
Reproducible vagrant and VM with openmodelica

### DISCUSSION THREAD

**Discussion by TomasKulhanek** [OpenModelica/OpenModelica#13185] [Show and tell]
Dear community,
for one of our project we needed a scriptable reproducible environment with OpenModelica. For this we use a `vagrant` tool that automates a lot of stuff needed when creating virtual machine, include port forwarding, SSH key generation and exchange, automatic VM bootstrap. 
I've found great Virtualbox VM images with dev last nightly Open Modelica and used them to create a so called `vagrant box`.

In order you already use 'vagrant' tool or want to start to use it, you may be based on the vagrant box 'openmodelica1.25' which contains nightly build of openmodelica 1.25 version. 
Required: 
- install vagrant tool (free install and download from https://www.vagrantup.com/) 
- install virtualbox

Then in command line (windows command line or linux terminal) following way:
```
vagrant init westlife-eu/openmodelica1.25
vagrant up
```
and after few minutes/hours to download and boot the VM image, you may enjoy it.

Comments, updates, recommendation welcomed.

**Comment by casella** [comment]
@adrpo, @arun3688, @AnHeuermann (just in case), would you like to try it?

**Comment by TomasKulhanek** [comment]
More complex workflow include editing the Vagrantfile in order to give more memory, cpu, one-time provision script etc:
```
vagrant init westlife-eu/openmodelica1.25
```
Then edit the created Vagrantfile and optionally put the following options `vb.gui=true` `vb.memory` and `vb.cpus`:
```
Vagrant.configure("2") do |config|
...
  config.vm.provider "virtualbox" do |vb|
     # Display the VirtualBox GUI when booting the machine
     vb.gui = true
     # Customize the amount of memory on the VM:
     vb.memory = "4096"
     # Set the number of CPUs to 2     
     vb.cpus = 2 
  end
  # one-time provisioning the VM guest
  config.vm.provision "shell", inline: <<-SHELL
    # the following is executed on guest using root privileges
    apt update
    apt upgrade -y openmodelica
  SHELL
...
end
```
then 
```
vagrant up 
```
As general, vagrant do port forwarding, by default ssh port `22` on guest -> `2200` on host. It shares file folder between host `(working dir where vagrant up)` to guest `/vagrant` folder. One time provisioning is possible adding a bootstrap script to Vagrantfile, e.g. to install other software or do additional upgrade from what is in the box image created months/years ago.
